Implement Server-Side analytics.

milkypostman opened the pull request:
@jiayliu @juberti

jiayliu commented on samples/web/content/apprtc/analytics.py:
```diff
@@ -0,0 +1,91 @@
+from oauth2client.appengine import AppAssertionCredentials
+from oauth2client.client import SignedJwtAssertionCredentials
+
+BIGQUERY_URL='https://www.googleapis.com/auth/bigquery'
```
put it in constants.py?

milkypostman commented:
Meant for this to be a PR against GoogleChrome. But I'm not sure I can do the comparison easily when it's to a different branch on my repo.

jiayliu commented on samples/web/content/apprtc/analytics.py:
```diff
@@ -0,0 +1,91 @@
+    # Attempt to initialize a connection to BigQuery.
+    self.bigquery = None
+    if is_running_locally:
+      # Local instances require a 'secrets.json' file.
```
Explain what secrets.json should look like and how to make one?

milkypostman replied:
I made googlechrome/webrtc#434 to add a `README` file. This information is a bit more appropriate there. I will add the docs to that when it is done.

jiayliu commented on samples/web/content/apprtc/analytics.py:
```diff
@@ -0,0 +1,91 @@
+    is_running_locally = os.environ['APPLICATION_ID'].startswith('dev')
+
+    self.bigquery_table = 'analytics'
+    self.bigquery_project = 'valiant-broker-834'
```
Put this in .yaml as a variable so that another project using our code can easily change it?
(https://cloud.google.com/appengine/docs/python/config/appconfig#Python_app_yaml_Defining_environment_variables)

milkypostman replied:
I am going to just make it use the `application` specified in `app.yaml`. Does that sound OK?

jiayliu replied:
Sounds OK.

jiayliu commented on samples/web/content/apprtc/analytics.py:
```diff
@@ -0,0 +1,91 @@
+    if is_running_locally:
+      self.bigquery_dataset = 'dev'
+    else:
+      self.bigquery_dataset = 'prod'
```
We may want some way to differentiate the data from different server versions, i.e. 1-dot-apprtc, 3-dot-apprtc. Can we pass that from the caller and use different dataset?

milkypostman replied:
You can do that but it seems a little too complex. I think what would be the right think to do would be to just store this information in the record. Is there some way for apprtc to know which server version it is?

jiayliu replied:
Yes, the http handler knows the host of the request, e.g. 1-dot-apprtc. See the use of get_memcache_key_for_room

jiayliu commented on samples/web/content/apprtc/analytics.py:
```diff
@@ -0,0 +1,91 @@
+
+BIGQUERY_URL='https://www.googleapis.com/auth/bigquery'
+
+class Analytics(object):
```
Add a unit test?

jiayliu commented on samples/web/content/apprtc/bigquery/analytics_schema.json:
```diff
@@ -0,0 +1,22 @@
+  {
+    "name": "impression_name",
+    "type": "string",
+    "mode": "nullable"
```
Why could this be nullable?

doep committed:
Add a README.md to apprtc directory.

The README initially contains how to run AppRTC locally.

Commit:
Move SDP manipulating url params parsing to client

jiayliu committed:
Merge pull request #443 from jiayliu/params

Move SDP manipulating url params parsing to client

doep committed:
Analytics class API for impressions to BigQuery

The Analytics class provides an API for sending impressions to
BigQuery. The bulk of the logic provided is to handle cases where apprtc
is run on AppEngine vs when it is running locally for dev.

When running on AppEngine, Analytics uses AppAssertionCredentials which
automatically provides valid credentials for the AppEngine project.

If running locally nothing is logged to BigQuery by default. In the case
that a 'secrets.json' file is provided all logging will be sent to
that project specified in the secrets file.

doep committed:
Add analytics to AppRTC server.

Logs an impression when two participants join a room.

jiayliu committed:
Merge pull request #431 from milkypostman/gapi-deps

Add Google API dependencies.

jiayliu committed:
Merge pull request #434 from milkypostman/add-readme

Add a README.md to apprtc directory.
